File: syntax.js
// equality check
var x = 10;
if (x == "10") {
  console.log("equal");
} else{
  console.log("not equal");
}

// falsey values
var x = 3;
if(null){
  x++;
}

if(false){
  x++;
}

if(0){
  x++;
}

if(""){
  x++;
}
console.log(x);


// scope
var x = 100;

function addOne(num) {
  x = 1;            //running this line changes the value of the original x variable. If you wanted (x = 100) to maintain it's value, you'd need to run (var x = 1) inside the function addOne
  return num + 1;
}

console.log(x);
console.log(addOne(5));
console.log(x);


// constructors

var boat = {
  name: "S.S. Minnow",
  color: "white",
  price: 2000,
  printInfo: function() {  //first citizen syntax. treating the function as a value
    console.log("The boat " + this.name + " has a price of " + this.price)
  }
};

boat.printInfo();

File: syntax.rb
# equality check
x = 10
if x == "10"
  puts "equal"
else
  puts "not equal"
end

# falsey values

x = 3
if nil
  x += 1
end

if false
  x += 1
end

if 0
  x += 1
end

if !!""
  x += 1
end

p x


# scope
x = 100

def add_one(num)
  x = 1
  return num + x
end

puts x
puts add_one(5)
puts x


# constructors

class Boat
  attr_reader :name, :color
  attr_writer :color

  def initialize(name, color, price)
    @name = name
    @color = color
    @price = price
  end

  def print_info
    puts "The boat #{@name} has a price of #{@price}"
  end
end

boat = Boat.new("SS Minnow", "white", 2000)
boat.print_info